### DISCUSSION TITLE
Receive push updates via Gotify

### DISCUSSION THREAD

**Discussion by jhotmann** [owntracks/android#1127] [Ideas & New Features]
Gotify is a great, self-hosted, push notification service. It also allows for custom json to be included with messages and the app sends a broadcast intent upon new messages:

- https://gotify.net/docs/msgextras
- https://github.com/gotify/android/blob/master/app/src/main/java/com/github/gotify/service/WebSocketService.java

These features would make it a great companion app for push updates from a server allowing the Owntracks app to use HTTP mode while still retaining real-time updates.

**Comment by growse** [comment]
Gotify looks like an interesting project. Am I right in thinking that if you have the Android client installed, then you can connect it to your own instance, and push messages to your device from your server?

What would Gotify allow you to do that isn't already done using MQTT-over-websockets?

**Reply by jhotmann** [reply]
The main benefit would be minimizing open connections. With things like [Unified Push](https://unifiedpush.org/) helping to boost Gotify's popularity in the self-hosted space, it could be a big battery saver for those already using Gotify.

If adding a broadcast receiver seems like too much, what about an intent that can be used to inject JSON payloads into the app? This would allow Tasker or other 3rd party apps to integrate data into the Owntracks map without needing to modify the Owntracks recorder to work with Gotify.

**Reply by growse** [reply]
If it's a simple case of adding a `BroadcastReceiver` that can react given certain types of messages, then I can see the advantages. 

We already have some intents that can be fired by Tasker to do certain actions (e.g. change mode, publish location etc.), but if there's any others that would be useful, we should get an issue raised to track those.
